lib/builtin.t

```diff
@@ -444,6 +444,50 @@ TODO: {
     is($HASH{key}, "val", 'Lexically exported hash is accessible');
 }
 
+# load_module
+{
+    use builtin qw( load_module );
+    use feature qw( try );
+    no warnings 'experimental::try';
+    my ($ok, $e);
+
+    # Failure to load module croaks
+    try {
+        load_module("Dies");
+        fail('load_module("Dies") succeeded!');
+    }
+    catch ($e) {
+        ok($e, 'load_module("Dies") fails');
+        like($e, qr/^Can't locate Dies[.]pm in \@INC/, 'load_module("Dies") fails with correct error');
+    }
+    my $module_name = 'Dies';
+    try {
+        load_module($module_name);
+        fail('load_module($module_name) $module_name=Dies succeeded!');
+    }
+    catch ($e) {
+        ok($e, 'load_module($module_name) $module_name=Dies fails');
+        like($e, qr/^Can't locate Dies[.]pm in \@INC/, 'load_module("Dies") fails with correct error');
+    }
+
+    # Loading module goes well
+    try {
+        load_module("strict");
+        pass('load_module("strict") worked');
+    }
+    catch ($e) {
+        fail('load_module("strict") errored: ' . $e);
+    }
+    $module_name = 'strict';
+    try {
+        load_module($module_name);
+        pass('load_module($module_name) $module_name=strict worked');
+    }
+    catch ($e) {
+        fail('load_module($module_name) $module_nam=strict errored: ' . $e);
+    }
+}
```

Comment:
Another test worth adding is the usual magic one; something like:

```perl
"Dies" =~ m/(\w+)/;
load_module($1);
...
```

and then check that it still behaves correctly. I believe your implementation currently won't, but it can be fixed by adding a call to `SvGETMAGIC(ST(0));` on a line before the `newSVsv`.

Reply:
I've added four of the type `load_module($1)`, two for failure and two for success: `$module_name =~ m!(\w+)!` and `"..." =~ m!(\w+)!` and all four seem to pass (on my machine) without needing to `SvGETMAGIC(...)`.  I'll watch test suite failures in case I can find other ways to reliably repro the requirement for having to add that call.

Reply:
Looks like those passed everywhere despite the lack of `SvGETMAGIC(..)`. Would you know why it should in principle be required, or have a suggestion for a test what would highlight it being required? Looks otherwise to not be required...

Reply:
Ooohright; yes. Because you're doing `newSVsv()` that already invokes get magic on its source argument.

In that case the code is fine but remember to add those tests in case of future edits.

Reply:
Tests for that are there:
- https://github.com/Perl/perl5/pull/21532/files#diff-3b3171c64e546f883b025fda758c78654dc5e73380370bf557acd6d9a10f7991R472-R489 for the failure to load a module (`$module_name = "Dies"` vs "Dies", then perform `=~` on either)
- https://github.com/Perl/perl5/pull/21532/files#diff-3b3171c64e546f883b025fda758c78654dc5e73380370bf557acd6d9a10f7991R507-R522 for the success (ditto, for `strict` instead of `Dies`